pages/docs/migration/migrating-to-v3.md

```diff
@@ -0,0 +1,393 @@
+
+In v3, the two operations are completely removed and replaced with an `action` property, that explicitly says what you (the application) do. Nothing about `others` and different perspectives to take into account.
+
+For more information about this publish and subscribe confusion here are some more reading materials:
+- Fran Méndez's [Proposal to solve publish/subscribe confusion](https://github.com/asyncapi/spec/issues/618)
+- Nic Townsend's blog post [Demystifying the Semantics of Publish and Subscribe](https://www.asyncapi.com/blog/publish-subscribe-semantics)
+
+Here is an example where, for simplicity, the application both consumes and produces messages to the test channel.
+
+```yml
+asyncapi: 2.6.0
+...
+channels: 
+  test/path:
+    subscribe:
+      ...
+    publish:
+      ...
+```
+
+```yml
+asyncapi: 3.0.0
+channels:
+  testPathChannel:
+    address: "test/path"
+    ...
+operations: 
+  publishToTestPath:
+    action: send
+    channel: 
+      $ref: "#/channels/testPathChannel"
+  consumeFromTestPath:
+    action: receive
+    channel: 
+      $ref: "#/channels/testPathChannel"
+```
+
+### Messages instead of message
+In v2, if you wanted to define channels to have one or more messages, you would do it with `oneOf`, or if just a single message.
+
+In v3, messages are now defined with the [Messages Object](https://www.asyncapi.com/docs/reference/specification/v3.0.0-next-major-spec.12#messagesObject), if you want a channel to have one or more messages, you just define multiple key/value pairs, or if a single message, it's just a single key/value pair.
+
+```yml
+asyncapi: 2.6.0
+...
+channels:
+  user/signedup:
+    message: 
+      oneOf:
+        - ...
+        - ...
+
+asyncapi: 2.6.0
+...
+channels:
+  user/signedup:
+    message: 
+      ...
+```
+
+```yml
+asyncapi: 3.0.0
+...
+channels:
+  UserSignup:
+    address: user/signedup
+    messages: 
+      UserMessage: 
+        ...
+      UserMessage2:
+        ...
+
+asyncapi: 3.0.0
+...
+channels:
+  UserSignup:
+    address: user/signedup
+    messages: 
+      UserMessage: 
+        ...
+```
+
+### Unifying explicit and implicit references
+
+In v2, it was possible to do implicit references in some places, for example for server security configuration, it was by name which referred to a [Security Schema Object](https://www.asyncapi.com/docs/reference/specification/v2.6.0#securitySchemeObject) in components - And for a channel to reference global servers by name.
+
+In v3, this information MUST be explicit references. This did mean that we had to slightly change the [Server Object](https://www.asyncapi.com/docs/reference/specification/v3.0.0-next-major-spec.12#serverObject) `security` property, which is now an array instead of an object. We then moved the information about needed scopes for OAuth and OpenID Connect to the [Security Scheme Object](https://www.asyncapi.com/docs/reference/specification/v3.0.0-next-major-spec.12#securitySchemeObject).
```

Comment:
```suggestion
In v3, this information MUST be explicit references. This did mean that we had to slightly change the [Server Object](https://www.asyncapi.com/docs/reference/specification/v3.0.0-next-major-spec.12#serverObject) `security` property, which is now an array instead of an object. We then moved the information about needed scopes for OAuth and OpenID Connect to the [Security Scheme Object](https://www.asyncapi.com/docs/reference/specification/v3.0.0-next-major-spec.12#securitySchemeObject).
```
Can we add links to website/docs for the `OAuth` and `OpenID Connect` is someone wants to know more about that?
Or are we assuming that the reader already knows what these are?

Reply:
We dont have any 🙂 

Then it would have to be external links to other resources, but it might be best just to let the user search for it then? 🤔

Reply:
> might be best just to let the user search for it then?
seems okay. Researching is important. :100: